prime-router/src/main/kotlin/serializers/Hl7Serializer.kt

```diff
@@ -1103,6 +1119,32 @@ class Hl7Serializer(val metadata: Metadata) : Logging {
         const val MESSAGE_TRIGGER_EVENT = "R01"
         const val SOFTWARE_VENDOR_ORGANIZATION: String = "Centers for Disease Control and Prevention"
         const val SOFTWARE_PRODUCT_NAME: String = "PRIME Data Hub"
-        val HD_FIELDS = listOf("MSH-4-1", "OBR-3-2", "OBR-2-2", "ORC-3-2", "ORC-2-2", "PID-3-4-1")
+
+        /*
+        From the HL7 2.5.1 Ch 2A spec...
+
+        The Hierarchical Designator identifies an entity that has responsibility for managing or
+        assigning a defined set of instance identifiers.
+
+        The HD is designed to be used either as a local identifier (with only the <namespace ID> valued)
+        or a publicly-assigned identifier, a UID (<universal ID> and <universal ID type> both valued)
+         */
+
+        /**
+         * List of fields that have the local HD type.
+         */
+        val HD_FIELDS_LOCAL = listOf("MSH-4-1", "OBR-3-2", "OBR-2-2", "ORC-3-2", "ORC-2-2",
+            "PID-3-4-1", "PID-3-6-1", "SPM-2-1-2", "SPM-2-2-2")
+
+        /**
+         * List of fields that have the universal HD type
+         */
+        val HD_FIELDS_UNIVERSAL = listOf("MSH-4-2", "OBR-3-3", "OBR-2-3", "ORC-3-3", "ORC-2-3",
+            "PID-3-4-2", "PID-3-6-2", "SPM-2-1-3", "SPM-2-2-3")
+
+        /**
+         * List of fields that have a CE type
+         */
+        val CE_FIELDS = listOf("OBX-15-1")
     }
```

Comment:
Same approch as before, but with little more metadata about the type of HL7 field.